### ISSUE TITLE
Merchant only having 1 trade that isn't able to be obtained as of the state of the village.

### ISSUE THREAD

**Issue by fdsfd1** [TangoTek/TekTopia-Community#275]
**Describe the bug**
My Merchant wants stuff I can't obtain, such as beetroot when I have wheat, potatoes, and carrots, or leather when I don't have a rancher and butcher, or jungle wood when I have oak and spruce, and the list goes on and on. Recently, I got attacked by a baby zombie that killed all my villagers except 2. Both of them were miners. After the incident, I haven't gotten a single miner trade.

**To Reproduce**
Honestly, I don't know how it happened. I'll say what I think it is.
1. Get 4-5 villagers.
2. Have a baby zombie with a heart in its hand that was made just before it spawned attack your villagers.
3. Make sure the only villager that survives is a miner.
4. Make the heart from the baby zombie a child.
5. Wait for the child to grow and make it a miner.
6. Look at the merchant trades for several ingame days.

**Expected behavior**
The merchant should have only miner trades as that is the only type of villager I have.

** Server/Location **
What server were you playing on? Main(Survival) or Dev (Creative)? Main 1
What are the world coordinates of your town hall marker? -908 66 1931

**Comment by JohnyDL** [comment]
This is right it's based on what you could have not what you do have for the beetroot and wood at least I don't know what's going on with the leather

Some odd behaviour anyway

**Comment by fdsfd-skripts** [comment]
>  
   
JohnyDL commented 12 minutes ago 
This is right it's based on what you could have not what you do have for the beetroot and wood at least I don't know what's going on with the leather
Some odd behaviour anyway

**Comment by fdsfd1** [comment]
> This is right it's based on what you could have not what you do have for the beetroot and wood at least I don't know what's going on with the leather
> Some odd behaviour anyway

This has been happening since yesterday when the attack happened.

**Comment by Sobertron** [comment]
you also should remember that you are still be offered the most basic trades, just not the ones you want. I would say there are three basic villagers: Farmer, Lumberjack, and Miner. Of those the former is probably the most basic, as they supply the food for your village. It isnt a stretch to say that the mod may assume that all villages will have farmers , as they are required to sustain your population, which may be why the merchant is offering you those trades.

**Comment by Tab-160** [comment]
I have been having the same problems, I have been getting several tool and armor trades even though I do not have a blacksmith.

**Comment by LinaBoeckwurm** [comment]
I've noticed that the paper trade is one of the earliest offered trades (together with the farmer, miner, and lumberjack trades), but honestly you don't get enchanter until fairly late into the game, and as far as I know the enchanter is the only one who converts sugarcane into paper.

**Comment by Konyv** [comment]
I have been having the same issue. 

I don't have a chicken scoop, pig pin, black Smith, or sheep pin, but the merchant will ask for diamond swords, eggs, and cooked meats. 

I've never had a baby zombie holding a heart. I assumed this was just a wide spread issue with Beta testing.
